modules/offload-s3/module.go

```diff
@@ -0,0 +1,278 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package modsloads3
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/peak/s5cmd/v2/command"
+	"github.com/peak/s5cmd/v2/log"
+	"github.com/peak/s5cmd/v2/log/stat"
+	"github.com/peak/s5cmd/v2/parallel"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+	"github.com/weaviate/weaviate/entities/modulecapabilities"
+	"github.com/weaviate/weaviate/entities/moduletools"
+	"github.com/weaviate/weaviate/usecases/config"
+)
+
+const (
+	Name        = "offload-s3"
+	s3Endpoint  = "OFFLOAD_S3_ENDPOINT"
+	s3Bucket    = "OFFLOAD_S3_BUCKET"
+	concurrency = "OFFLOAD_S3_CONCURRENCY"
+	timeout     = "OFFLOAD_TIMEOUT"
+)
+
+// verify we implement the modules.Module interface
+var (
+	_ = modulecapabilities.Module(New())
+)
+
+// we use custom cli app to avoid some bugs in underlying dependencies
+// specially with .After implementation.
+var app = &cli.App{
```

Comment:
Can this app be created together with module in `New()` method? why it's being creating separately?

Reply:
good catch, it was just for my local testing ,makes total sense to add it to`New()`